### PR TITLE
(Also PAUSED) 9561 - Implement pagination on Transaction History

### PR DESCRIPTION
**Pull request by robbiecarlton**
After discussions with @pjkundert about how HoloFuel DNA pagination works, I've decided to go with local pagination for MVP. We'll revisit this after launch.

### CODE REVIEW

File: src/data-interfaces/HoloFuelDnaInterface.js [hunk 1]
```diff
@@ -252,6 +252,7 @@ const HoloFuelDnaInterface = {
     },
     allDeclinedTransactions: async () => {
       const declinedResult = await createZomeCall('transactions/list_pending_declined')()
+      console.log('declinedResult', declinedResult)
```
Comment (JettTech) [inline src/data-interfaces/HoloFuelDnaInterface.js]: Delete log. :wink:

File: src/mock-dnas/holofuel.js [hunk 2]
```diff
@@ -421,6 +485,7 @@ const holofuel = {
     ledger_state: () => transactionList.ledger,
     list_transactions: () => transactionList,
     list_pending: listPending,
+    list_pending_declined: [],
```
Comment (JettTech) [inline src/mock-dnas/holofuel.js:488]: :+1:

File: src/holofuel/pages/TransactionHistory/TransactionHistory.js [hunk 3]
```diff
@@ -123,6 +128,8 @@ export default function TransactionsHistory () {
       </React.Fragment>)}
     </div>}
 
+    {hasMoreTransactions && <Button onClick={showMoreTransactions}>Load More</Button>}
```
Comment (JettTech) [inline src/holofuel/pages/TransactionHistory/TransactionHistory.js:131]: Let's add the conditional that omits this button component whenever there are no transactions.
ie: `{!noVisibleTransactions && hasMoreTransactions && <Button onClick={showMoreTransactions}>Load More</Button>}`

**Review by JettTech — CHANGES_REQUESTED** [review]
I have requested a few touch ups, but looks good for an interim pagination approach.  I'd also like to talk over this new approach and understand why Perry said to avoid the use of the hf dna pagination logic.

File: src/holofuel/pages/TransactionHistory/TransactionHistory.js (hunk 3 above)
Comment (robbiecarlton) [reply src/holofuel/pages/TransactionHistory/TransactionHistory.js:131]: awkwardly, it's possible for there to be more transactions to load, even while there are `noVisibleTransactions`. (eg: there are no withdrawals in the current transaction list, but there are some later.) This is a slightly broken UX, and we can make it smarter, but I want to get a first iteration in before we ship.

**Pull request closed without merging**